### ISSUE TITLE
[FEAT] Dataset EDA

### ISSUE THREAD

**Issue by jerry-ryu** [jerry-ryu/Seg_Depth#1]
# Overview
 NYU 및 KITTI data에 대해서 현재 baseline(AdaBins)이 1. 얼마나 이해하고 있는지 2. 잘 학습하지 못한 부분은 무엇인지 찾기 위해 EDA가 필요하였다.

# Todo
### Image, depth map visualization
> - [x] RGB 이미지 viz.
> - [x]  Pred depth map, GT depth map viz
> - [x]  Error map viz.
 

> 문제 발생:
>1. Pred depth map과 GT depth map의 color bar 차이로 인해 고정 값 normalize 적용 -> 고정 값보다 깊거나 얕은 경우 깊이감이 사라짐

```
# infer_comapre.py
gt_cmap = plt.get_cmap('plasma')
gt_norm = Normalize(vmin=0, vmax=5000) # normalize 0~5000, 5000 이상 값은 깊이감 x
```
>- [x]  고정 / 변동 colorbar Pred, GT depth map 각각 viz.
>2. uniform한 분포를 Adabins의 분포와 비교하기 어려움
>- [x]  Adabins hist의 bin개수(20)와 맞춘 uniform bins plot / args.n_bins와 맞춘 uniform bins plot 추가
>3.  Adabins으로 pred depth map을 hist 해야 GT hist와 비교 가능
>- [x]  pred depth map을 Adabins으로 hist.한 plot 추가

### Metric logging
> - [x] 이미지마다 depth metric('a1','a2', 'a3, 'abs_rel', 'rmse', 'log_10', 'rmse_log', 'silog', 'sq_rel') logging
> - [x] 이미지마다 AdaBins가 histogram을 잘 따르고 있는지 chamfer loss logging
> - [x] 이미지마다 depth hist. with (AdaBins, Uniform bins) 그리고 Adabins의 hist. logging
> - [x] 전체 데이터셋의 metric logging(metric.txt)
> - [x] 각 metric CSV로 저장(metric.csv)하여 필터링 가능

# Additional Context
EDA example:
1. NYU:
![rgb_00076](https://github.com/jerry-ryu/Seg_Depth/assets/62556539/f81fa767-df0e-4ce4-8102-c9f5194a0ad9)
2. Kitti:

> 문제 발생:
> 1. 'online_eval' dataloader를 사용하여 batch_size 1로 고정 -> NYU train dataset EDA에 12시간 소요,
>- [x]  배치 사이즈를 늘릴 수 있게 코드 수정 필요

**Comment by jerry-ryu** [comment]
AdaBins의 test 데이터를 적어놓은 
AdaBins/train_test_inputs/kitti_eigen_test_files_with_gt.txt에 GT 데이터가 누락된 경우 발견 후 제외

> ex. 2011_09_26/2011_09_26_drive_0002_sync/image_02/data/0000000003.png None 721.5377

원래 kitti_eigen_test_files_with_gt.txt => 데이터 개수: 697
수정후 kitti_eigen_test_files_with_gt_edit.txt => 데이터 개수:  652

+)
Kitti 자체에 depth GT가 없는 것 같음
https://github.com/shariqfarooq123/AdaBins/issues/10
